### DISCUSSION TITLE
[Bug]: mne with vscode permissions error on Nectar Desktop

### DISCUSSION THREAD

**Discussion by air2310** [neurodesk/neurodesk.github.io-hugo#274] [Bug Report]
### Version

Neurodesk through Nectar research cloud virtual desktop service (not sure what version is running)

### What platform are you seeing the problem on?

Nectar

### What happened

Opening any of the mne/vscode containers results in the following error from vscode:

""Unable to write program user data. A system error occurred. Please make sure the following directories are writable: 
~/.confic/Code
/opt/vscode-extensions
/neurodesktop-storage"

It seems as though the opt directory is indeed not writable, so that might be the route of the issue. 

Screenshot of error attached:
![Capture](https://user-images.githubusercontent.com/20147427/235947623-e09a9f3f-91cd-4379-8d2f-8c145659691c.PNG)

### How to reproduce

Open up any of the mne/code containers through the virtual desktop service.

**Comment by stebo85** [comment]
Dear @air2310,

yes, that's indeed a bug. The problem comes from a missing /neurodesktop-storage bind mount.

The problem exists in the cloud version of neurodesk and on nectar

To fix it in nectar:
```
vi ~/.bashrc
```
Then add this at the end of the file:
```
export neurodesk_singularity_opts="${neurodesk_singularity_opts} --bind /neurodesktop-storage "
```

I will fix this now in nectar.







To fix it in the cloud version:
```
vi ~/.bashrc
```
Then change this:
```
export neurodesk_singularity_opts="${neurodesk_singularity_opts} --bind /home/jovyan:/home/matlab/.matlab/R2022a_licenses,/home/jovyan:/opt/matlab/R2022a/licenses "
```

To:
```
export neurodesk_singularity_opts="${neurodesk_singularity_opts} --bind /neurodesktop-storage,/home/jovyan:/home/matlab/.matlab/R2022a_licenses,/home/jovyan:/opt/matlab/R2022a/licenses "
```

This will be fixed with the new cloud image that @aswinnarayanan is working on and should be ready in a few days.

Thank you for finding this problem :)
Steffen

**Reply by stebo85** [reply]
There is also another little bug in the mne  container that tries to execute module load inside the container when opening a terminal. For this the fix is:

replace:
```
if [ -d /cvmfs/neurodesk.ardc.edu.au/neurodesk-modules ]; then
        # export MODULEPATH="/cvmfs/neurodesk.ardc.edu.au/neurodesk-modules"
        module use /cvmfs/neurodesk.ardc.edu.au/neurodesk-modules/*
else
        export MODULEPATH="/neurodesktop-storage/containers/modules"              
        module use $MODULEPATH
        export CVMFS_DISABLE=true
fi


if [ -f '/usr/share/module.sh' ]; then
        echo 'Run "ml av" to see which tools are available - use "ml <tool>" to use them in this shell.'
        if [ -v "$CVMFS_DISABLE" ]; then
                if [ ! -d $MODULEPATH ]; then
                        echo 'Neurodesk tools not yet downloaded. Choose tools to install from the Application menu.'
                fi
        fi
fi
```

with
```
if [ -f '/usr/share/module.sh' ]; then
        if [ -d /cvmfs/neurodesk.ardc.edu.au/neurodesk-modules ]; then
                module use /cvmfs/neurodesk.ardc.edu.au/neurodesk-modules/*
        else
                export MODULEPATH="/neurodesktop-storage/containers/modules"
                module use $MODULEPATH
                export CVMFS_DISABLE=true
        fi

        echo 'Run "ml av" to see which tools are available - use "ml <tool>" to use them in this shell.'
        if [ -v "$CVMFS_DISABLE" ]; then
                if [ ! -d $MODULEPATH ]; then
                        echo 'Neurodesk tools not yet downloaded. Choose tools to install from the Application menu.'
                fi
        fi
fi
```

**Reply by air2310** [reply]
Thanks @stebo85! I somehow didn't see the github notification about your response to this, so only just found it, but it will still be super helpful to me in the coming days :) If the new cloud image is up and running, I can let you know if there's any further issues with it soon.
